### ISSUE TITLE
And ManifoldCAD?

### ISSUE THREAD

**Issue by elalish** [cookiecad/Openscad-to-JSCAD-converter#10]
So excited to see this project! JSCAD and [ManifoldCAD](https://manifoldcad.org/) bear much resemblance - might it be possible to translate to our API? I've actually been considering releasing our [worker](https://github.com/elalish/manifold/blob/master/bindings/wasm/examples/worker.ts) code on npm to make it easy for anyone to take manifold-style JS/TS scripts and generate GLB/3MF models. This OpenSCAD conversion would give that a lot more value. 

FYI @pca006132, since we've occasionally chatted about how to make a converter like this.

**Comment by napter** [comment]
Yes, definitely. I'll upload a POC soon. It won't be terribly difficult to change the jscad generator to a manifold generator.

**Comment by napter** [comment]
@elalish I've added the scaffolding for Manifold. It can now convert all manner of openscad cubes (below) to manifoldcad:

```
cube();
cube(1);
cube(size=2);
cube(3,true);
cube(size=4, center=true);
cube([1,2,3]);
cube([2,3,4],true);
cube(size=[3,4,5]);
cube(size=[4,5,6], center=true);
```

**Comment by napter** [comment]
@elalish Can you see if anyone in your community wants to help out in writing the manifold syntax generator? That would be really helpful in getting this off the ground. 

I'll spin up a hosted version shortly so  once it is useful, people don't have to download the repo to do the conversions.

**Comment by napter** [comment]
@elalish not a lot of variety yet, but here is a PR implementing cylinder: https://github.com/cookiecad/Openscad-to-JSCAD-converter/pull/11

**Comment by napter** [comment]
@elalish is there documentation that covers the manifold javascript syntax? I was having trouble locating it - the link from the npm package is 404.

Can you also please point me to the code where the objects and functions are all declared? 

Thanks!

**Comment by elalish** [comment]
Yes, sorry we just fixed that, but that page won't update till we release a new npm version. Here's our brand new TS docs page: https://manifoldcad.org/jsdocs/

**Comment by napter** [comment]
@elalish great! Those pages look like they are generated by something. Do you have the data that built those pages in a structured format? If you can send that over that could be very helpful.

**Comment by elalish** [comment]
They're from our TS type [declarations](https://github.com/elalish/manifold/blob/master/bindings/wasm/manifold-encapsulated-types.d.ts).
